test/jdk/com/sun/java/swing/plaf/gtk/TestFileChooserSingleDirectorySelection.java

```diff
@@ -48,130 +47,130 @@
 public class TestFileChooserSingleDirectorySelection {
     private static JFrame frame;
     private static JFileChooser fileChooser;
-    private static JButton getSelectedFilesButton;
     private static Robot robot;
-    private static boolean passed;
-    private static File[] testDir;
-    private static File[] tempFile;
+    private static volatile Point frameLocation;
+    private static volatile int frameWidth;
+    private static volatile int frameHeight;
 
     public static void main(String[] args) throws Exception {
         System.setProperty("sun.java2d.uiScale", "1.0");
         robot = new Robot();
         robot.setAutoDelay(100);
-
-        try {
-            // create test directory
-            String tmpDir = System.getProperty("user.home");
-            testDir = new File[1];
-            testDir[0] = new File(tmpDir, "testDir");
-            if (!testDir[0].exists())
-                testDir[0].mkdir();
-            testDir[0].deleteOnExit();
-
-            // create temporary files inside testDir
-            tempFile = new File[5];
-            for (int i = 0; i < 5; ++i) {
-                tempFile[i] = File.createTempFile("temp", ".txt",
-                        new File(testDir[0].getAbsolutePath()));
-                tempFile[i].deleteOnExit();
-            }
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+        File testDirDirs = createFoldersOnlyDir();
+        File testDirFiles = createFilesOnlyDir();
+        populateDirs(testDirDirs);
+        populateFiles(testDirFiles);
         for (UIManager.LookAndFeelInfo laf :
-                        UIManager.getInstalledLookAndFeels()) {
+                UIManager.getInstalledLookAndFeels()) {
             System.out.println("Testing LAF: " + laf.getClassName());
             SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
-            checkFileOnlyTest(laf);
-            checkDirectoriesOnlyTest(laf);
-            checkFilesAndDirectoriesTest(laf);
+            checkFileOnlyTest(laf, testDirFiles);
+            checkDirectoriesOnlyTest(laf, testDirDirs);
+            checkFilesAndDirectoriesTest(laf, testDirDirs);
```

Comment:
So, `FILES_AND_DIRECTORIES` mode is tested with directories only. Is it intentional? It's the problem in [JDK-6972078](https://bugs.openjdk.org/browse/JDK-6972078)… that directories can't be selected.

Reply:
> So, `FILES_AND_DIRECTORIES` mode is tested with directories only. Is it intentional? It's the problem in [JDK-6972078](https://bugs.openjdk.org/browse/JDK-6972078)… that directories can't be selected.

Yes, it is tested with directories only. 

> Is it intentional? It's the problem in [JDK-6972078](https://bugs.openjdk.org/browse/JDK-6972078)… that directories can't be selected.

Not intentional though. Thought of creating a single folder with the mix of files and directories initially but getting the location of files in `FILES_ONLY` mode was an issue in different LAF. So, created separate folders for `directories` and `files` and tested.

Reply:
> > Is it intentional? It's the problem in [JDK-6972078](https://bugs.openjdk.org/browse/JDK-6972078)… that directories can't be selected.
> 
> Not intentional though. Thought of creating a single folder with the mix of files and directories initially but getting the location of files in `FILES_ONLY` mode was an issue in different LAF. So, created separate folders for `directories` and `files` and tested.

It's fine as long as the original issue is reproducible with directories only too.